includes/class-wporg-notifications.php

```diff
@@ -429,12 +429,9 @@ private static function get_original( int $post_id ) {
 	 * @return array The list of emails with the opt-in enabled.
 	 */
 	private static function get_opted_in_email_addresses( array $email_addresses ): array {
-		foreach ( $email_addresses as $email_address ) {
-			if ( self::is_global_optout_email_address( $email_address ) ) {
-				$index = array_search( $email_address, $email_addresses, true );
-				if ( false !== $index ) {
-					unset( $email_addresses[ $index ] );
-				}
+		foreach ( $email_addresses as $index => $email_address ) {
+			if ( ! $email_address || self::is_global_optout_email_address( $email_address ) ) {
```

Comment:
Could we leave the function signature below the same and check here for a string?

```suggestion
			if ( ! is_string( $email_address ) || empty( $email_address ) || self::is_global_optout_email_address( $email_address ) ) {
```